File: app/src/main/java/com/lazylee/lzywanandroid/tools/FragmentHelper.java
package com.lazylee.lzywanandroid.tools;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import java.util.List;

/**
 * fragment helper
 * Created by lazylee on 2018/4/9.
 */

public class FragmentHelper {

    public static Fragment getCurrentVisibleFragment(@NonNull FragmentManager manager){
        List<Fragment> fragments = manager.getFragments();
        if (fragments != null && !fragments.isEmpty()) {
            for (Fragment fragment : fragments) {
                if (fragment != null && fragment.isVisible()) {
                    return fragment;
                }
            }
        }
        return null;
    }

    @Nullable
    public static Fragment getFragmentByTag(@NonNull FragmentManager fragmentManager, @NonNull String tag) {
        return fragmentManager.findFragmentByTag(tag);
    }
}

File: app/src/main/java/com/lazylee/lzywanandroid/tools/FragmentHelper.kt
package com.lazylee.lzywanandroid.tools


import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

/**
 * fragment helper
 * Created by lazylee on 2018/4/9.
 */

object FragmentHelper {

    fun getCurrentVisibleFragment(manager: FragmentManager): Fragment? {
        val fragments = manager.fragments
        if (fragments != null && !fragments.isEmpty()) {
            for (fragment in fragments) {
                if (fragment != null && fragment.isVisible) {
                    return fragment
                }
            }
        }
        return null
    }

    fun getFragmentByTag(fragmentManager: FragmentManager, tag: String): Fragment? {
        return fragmentManager.findFragmentByTag(tag)
    }
}
